satellite/gracefulexit/endpoint.go

```diff
@@ -182,6 +182,12 @@ func (endpoint *Endpoint) doProcess(stream processStream) (err error) {
 	pending := newPendingMap()
 
 	var morePiecesFlag int32 = 1
+	var errorFlag int32 = 0
+	handleError := func(err error) error {
+		atomic.StoreInt32(&errorFlag, 1)
```

Comment:
I guess I'm curious what's the benefit of using the atomic store vs. just returning the wrapped error?

Reply:
I added this so I didn't have to call group.Wait() in the for loop to know if an error occurred. But there's a probably a better way I'm just not familiar with in Go.

Reply:
maybe we could have an error channel? I'm not totally sure what would be better either- maybe this is fine, but I just don't think I'd seen it used before.

Reply:
I agree with having an error channel

Reply:
Updated to use channel.